app/controllers/api/v1/sessions_controller.rb

```diff
@@ -28,7 +28,12 @@ def create_params
 
   def create_session_for_user
     ensure_one_active_session
-    @session = Session.create!(create_params)
+    pending_referral = Referral.pending.for_candidate(current_user).first
+    create_params[:rewards] = pending_referral.reward if pending_referral
```

Comment:
Can we keep "rewards due to referrals" and "rewards due to sessions" as seperate things?
Or else by looking at a session object it would be hard to know if the reward has come from a referral or by staying online at home.